rust/feature-flags/src/flag_matching.rs

```diff
@@ -805,6 +824,85 @@ impl FeatureFlagMatcher {
         }
     }
 
+    pub async fn evaluate_cohort_filters(
+        &self,
+        filters: &[PropertyFilter],
+        target_properties: &HashMap<String, Value>,
+    ) -> Result<bool, FlagError> {
+        Box::pin(self.evaluate_potentially_nested_cohort_filters(filters, target_properties)).await
+    }
+
+    async fn evaluate_potentially_nested_cohort_filters(
+        &self,
+        filters: &[PropertyFilter],
+        target_properties: &HashMap<String, Value>,
+    ) -> Result<bool, FlagError> {
+        let mut cohort_filters = Vec::new();
+        let mut non_cohort_filters = Vec::new();
+
+        // Separate cohort filters from non-cohort filters
+        for filter in filters {
+            if filter.prop_type == "cohort" {
+                cohort_filters.push(filter);
+            } else {
+                non_cohort_filters.push(filter);
+            }
+        }
+
+        // Evaluate non-cohort filters
+        for filter in &non_cohort_filters {
+            if !match_property(filter, target_properties, false).unwrap_or(false) {
+                return Ok(false);
+            }
+        }
+
+        // Evaluate cohort filters
+        if !cohort_filters.is_empty() {
+            let cohort_ids: HashSet<CohortId> = cohort_filters
+                .iter()
+                .filter_map(|f| f.value.as_i64().map(|id| id as CohortId))
```

Comment:
`value` needs to be `i64` but then I cast it to match the DB type.  It's dumb but safe enough.